lib/rules/self-closing-void-elements.js

```diff
@@ -83,16 +85,30 @@ export default class SelfClosingVoidElements extends Rule {
               ? `${source.slice(0, -1)}/>`
               : `${source.slice(0, -2)}>`;
 
-            this.log({
-              message: isSelfClosingRequired
-                ? 'Self-closing a void element is required'
-                : 'Self-closing a void element is redundant',
-              node,
-              source,
-              fix: {
-                text: expected,
-              },
-            });
+            if (this.mode === 'fix') {
+              let editedNode = builders.element(node.tag, {
```

Comment:
Are you sure we can't use the existing node instead of creating a new one? Wouldn't that be easier? And then we wouldn't have to worry about remembering to pass all these arguments to it.

Reply:
I tried setting `selfClosing` but it wasn't doing anything. We could also do `{ ...node, attrs: node.attributes }`, but I didn't want to pass too much, in case the API adds options.

Reply:
Hmm, okay, that's weird, I'm not sure how we can trigger it to update after just modifying that boolean property.

So we can stick with your current approach, but we need to make sure all these arguments are tested then. So you'll need to update at least one test case to add all these arguments in the `template` and make sure that they are copied over to the `fixedTemplate`.

Reply:
So I was working a bit more on this and I'm wondering. Attributes and modifiers are fine, but self-closing tags can't really have block params or children right? They're a leaf node.

Reply:
My guess is you don't need to worry about that as only the relevant data will be rendered depending on how `selfClosing` is set. So try just copying everything over to the new node.

Reply:
Looks like all we need to finish this PR is to add some test cases that demonstrate everything is copied from the old node to new node correctly.